azure/scope/machinepoolmachine.go

```diff
@@ -281,6 +294,7 @@ func (s *MachinePoolMachineScope) UpdateStatus(ctx context.Context) error {
 	}
 
 	if err != nil && !apierrors.IsNotFound(err) {
+		conditions.MarkFalse(s.AzureMachinePoolMachine, clusterv1.MachineNodeHealthyCondition, clusterv1.NodeNotFoundReason, clusterv1.ConditionSeverityError, "")
```

Comment:
@CecileRobertMichon @devigned @mboersma Same with here

Reply:
This is if `!apierrors.IsNotFound(err)`, so it's _not_ a NodeNotFound reason. It's that we failed to find the node because one of those func returned an error. Maybe we need to add an else statement if the node is actually not found, then we can mark the condition as NotFound. In this case, I think `WaitingForNodeRefReason` makes the most sense.

Reply:
I think we need to split conditions for the two scenarios above:

`if nodeRef == nil || nodeRef.Name == "" `: Node ref doesn't exist yet, node could still be provisioning - `NodeProvisioningReason`
`else`:  node ref is set, but node is not found: `NodeNotFoundReason`